### PR TITLE
Release 1.2.0

### PR DESCRIPTION
**Pull request by mandar242**
1. Updated the version number to `1.2.0`
2. Ran `go generate` to regenerate docs
3. Ran `antsibull-changelog release --version 1.2.0`

### CODE REVIEW

**Comment by gravesm** [comment]
Could you reissue this against the stable-1 branch? We should do releases from that.

**Review by abikouo — APPROVED** [review]

**Comment by gravesm** [comment]
@sikha-root Thanks for catching this. Yes, we need to update the version in go.mod.

**Comment by gravesm** [comment]
Sure, if you wouldn't mind creating a PR to update this, that would be great. I'd like to also update https://github.com/ansible/terraform-provider-ansible/blob/e5e644864010e43ca53bcb37cbc8b04e5e9bc871/.github/workflows/integration.yml#L30 to use `go-version-file: 'go.mod'`. This will ensure the same version is being used to run integration tests as the version being used to build releases. This way we can catch these problems sooner.

**Comment by gravesm** [comment]
This PR appears to still be based off of the main branch, which is why it's duplicating commits that have been backported to the stable-1 branch. The commit that you've made should be rebased on the stable-1 branch.

**Review by hakbailey — APPROVED** [review]

File: CHANGELOG.rst [hunk 1]
```diff
@@ -0,0 +1,19 @@
+================================================
+The Terraform Provider for Ansible Release Notes
+================================================
+
+.. contents:: Topics
+
+v1.2.0
+======
+
```
[the remark this answers is not in the document]
Comment (sikha-root) [reply CHANGELOG.rst:9]: I didn't include a fragment in the library update (wasn't familiar with antsibull when I submitted the PR) but I'd also suggest adding an extra minor_change for it too: https://github.com/317project/terraform-provider-ansible/commit/3b134a85aaf691ab23bad94cbacc29dd75826e93

File: changelogs/changelog.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    changes:
+      bugfixes:
+      - provider/resource_playbook - Fix race condition between multiple ansible_playbook
+        resources (https://github.com/ansible/terraform-provider-ansible/issues/38).
+      minor_changes:
+      - Update dependencies (google.golang.org/grpc and golang.org/x/net) to resolve
+        security alerts https://github.com/ansible/terraform-provider-ansible/security/dependabot
+        (https://github.com/ansible/terraform-provider-ansible/pull/72).
+      - Updates the provider to use SDKv2 (https://github.com/ansible/terraform-provider-ansible/issues/39).
+    fragments:
```
Comment (GomathiselviS) [inline changelogs/changelog.yaml:13]: Please include `release_summary` here as well.

File: CHANGELOG.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (mandar242) [reply CHANGELOG.rst:9]: @alinabuzachis should the changelog fragment for #89 go in that/another PR?
or can I add the changelog to this #87  (release) PR itself?

File: changelogs/changelog.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    - aws_example.yml
+    - gplv3_licensing.yml
+    - inventory_race_conditions.yml
+    - update_dependencies.yml
+    - use_sdkv2.yml
+    release_date: '2024-02-19'
```
[the remark this answers is not in the document]
Comment (mandar242) [reply changelogs/changelog.yaml]: fixed

File: changelogs/changelog.yaml (hunk 2 above)
Comment (mandar242) [reply changelogs/changelog.yaml:15]: fixed

File: CHANGELOG.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (alinabuzachis) [reply CHANGELOG.rst:9]: @mandar242 I would lean towards manually backporting that PR separately, then rebase this and regenerate the changelog. I think it's best to keep it focused only on the purposes of the release without including any further changes.

[the remark this answers is not in the document]
Comment (sikha-root) [reply CHANGELOG.rst:9]: That PR was already backported into stable-1 (#91), hence my suggestion. It just didn't have a fragment that `antsibull-changelog` could pick up.

[the remark this answers is not in the document]
Comment (mandar242) [reply CHANGELOG.rst:9]: Added changelog for #89

**Review by alinabuzachis — APPROVED** [review]

**Pull request merged**